src/explore-education-statistics-admin/src/pages/release/ReleaseMetaGuidancePage.tsx

```diff
@@ -48,11 +52,22 @@ const ReleaseMetaGuidancePage = ({
               caption={model.release.title}
             />
 
-            <h2>Metadata guidance document</h2>
+            <h2>Data guidance</h2>
 
             <ReleaseMetaGuidancePageContent
```

Comment:
We should also rename all of the `MetaGuidance*` components across the project as well